### PR TITLE
Linting Updates

### PR DESCRIPTION
**Pull request by cwild-UoS**
Originally was going to solve more of the linting issues but I think it would be more appropriate on the other side of the discussed refactor

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -151,7 +151,9 @@ disable=raw-checker-failed,
         suppressed-message,
         useless-suppression,
         deprecated-pragma,
-        use-symbolic-message-instead
+        use-symbolic-message-instead,
+        invalid-name,
```
Comment (bobturneruk) [inline .pylintrc]: It is possible I think to set this up to check for either snake_case or a single character string, which I think would cover all of our variables, via a regex. Although I'm not sure why the single characters are being complained about by pylint as I think single characters *are* valid snake case.

File: .pylintrc [hunk 2]
```diff
@@ -151,7 +151,9 @@ disable=raw-checker-failed,
...
+        unnecessary-pass
```
Comment (bobturneruk) [inline .pylintrc]: We don't need this - I think some pass statements are necessary, some aren't. pylint can tell. I've pushed a commit ([001fedc](https://github.com/CITCOM-project/CausalTestingFramework/pull/99/commits/001fedc3d448489b36a1f29c20506e7b9e3b463a)) so hopefully I'm not wrong.

File: causal_testing/data_collection/data_collector.py [hunk 3]
```diff
@@ -73,7 +73,10 @@ def filter_valid_data(self, data: pd.DataFrame, check_pos: bool = True) -> pd.Da
         size_diff = len(data) - len(satisfying_data)
         if size_diff > 0:
             logger.warning(
-                f"Discarded {size_diff}/{len(data)} values due to constraint violations.\n" f"For example{unsat_core}"
+                "Discarded %s/%s values due to constraint violations.\n" "For example%s",
```
Comment (bobturneruk) [inline causal_testing/data_collection/data_collector.py:75]: Looking at these [w1203](https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html) things, I prefer them as F-strings. I think the readability of F-strings outweighs the performance / other potential benefits of %. More discussion here https://github.com/PyCQA/pylint/issues/2395

Comment (bobturneruk) [reply causal_testing/data_collection/data_collector.py:75]: This would mean disabling some checks for a clean linter run.

**Review by bobturneruk — APPROVED** [review]
Thanks for looking at this - I have a few suggestions.

File: causal_testing/data_collection/data_collector.py (hunk 3 above)
Comment (cwild-UoS) [reply causal_testing/data_collection/data_collector.py:75]: I agree as well, F-Strings are much clearer and I'm not sure there would be a noticeable performance impact. So I'll happily revert them to F-Strings for now!

Interesting read as well cheers Bob :)

File: .pylintrc (hunk 2 above)
Comment (cwild-UoS) [reply .pylintrc]: Ahh okay, I see that it is unnecessary when there are docstrings present as it's not 'empty code'. Personally I like them as they imply that the method is incomplete / needs filling in. I'm happy either way and to go with what the linter wants.

File: .pylintrc (hunk 1 above)
Comment (cwild-UoS) [reply .pylintrc]: I've added a regex to catch snake_case of 1 or more character variable names now

**Pull request merged**